frontend/src/components/pages/teacher/DisplayClassroomPage/index.tsx

```diff
@@ -0,0 +1,115 @@
+import React from "react";
+import { Redirect, useHistory, useLocation, useParams } from "react-router-dom";
+import { Flex, IconButton, Tag } from "@chakra-ui/react";
+
+import { EditOutlineIcon } from "../../../../assets/icons";
+import * as Routes from "../../../../constants/Routes";
+import {
+  formatMonth,
+  removeUnderscore,
+  titleCase,
+} from "../../../../utils/GeneralUtils";
+import HeaderWithButton from "../../../common/HeaderWithButton";
+import FormBreadcrumb from "../../../common/navigation/FormBreadcrumb";
+import RouterTabs from "../../../common/navigation/RouterTabs";
+import AddClassroomOrStudentPopover from "../../../teacher/student-management/classrooms/AddClassroomOrStudentPopover";
+import NotFound from "../../NotFound";
+
+import DisplayClassroomAssessmentsPage from "./DisplayClassroomAssessmentsPage";
+import DisplayClassroomStudentsPage from "./DisplayClassroomStudentsPage";
+
+const TAB_CONFIG = [
+  {
+    name: "Assessments",
+    path: Routes.DISPLAY_CLASSROOM_ASSESSMENTS_PAGE(),
+    Component: DisplayClassroomAssessmentsPage,
+  },
+  {
+    name: "Students",
+    path: Routes.DISPLAY_CLASSROOM_STUDENTS_PAGE(),
+    Component: DisplayClassroomStudentsPage,
+  },
+  {
+    path: Routes.DISPLAY_CLASSROOM_PAGE(),
+    exact: true,
+    Component: () => {
+      const { classroomId } = useParams<{ classroomId: string }>();
+      const location = useLocation();
+      return (
+        <Redirect
+          to={{
+            pathname: Routes.DISPLAY_CLASSROOM_STUDENTS_PAGE(classroomId),
+            state: location.state,
+          }}
+        />
+      );
+    },
+  },
+  {
+    path: "*",
+    Component: NotFound,
+  },
+];
+
+const BREADCRUMB_CONFIG = (className: string) => [
+  {
+    header: "Classrooms",
+    page: 0,
+  },
+  {
+    header: className,
+    page: 1,
+  },
+];
+
+const getLocationState = (
+  state: unknown,
+): { className?: string; startDate?: string; grade?: string } => ({
+  className: undefined,
+  startDate: undefined,
+  grade: undefined,
+  ...(typeof state === "object" ? state : {}),
+});
+
+const DisplayClassroomsPage = () => {
+  const history = useHistory();
+  const { state } = useLocation();
+  const { className, startDate, grade } = getLocationState(state);
+
+  const classTitle = className;
+  const effectiveTitle = classTitle ?? "Loading...";
```

Comment:
instead of passing in a fixed `Loading..` constant, should we just update the bread config to have a loading state?

Reply:
good idea